core/src/main/java/org/infinispan/marshall/persistence/impl/PersistenceMarshallerImpl.java

```diff
@@ -53,6 +53,8 @@
    private static final Log log = LogFactory.getLog(PersistenceMarshallerImpl.class, Log.class);
 
    @Inject GlobalComponentRegistry gcr;
+
+   //todo [anistor] Do we need a WrappedMessageTypeMapper here? Also, there is another SerializationContext instance in ProtobufMetadataManagerImpl
```

Comment:
@ryanemerson we have 2 SerializationContext instances. Should we share them ? Should both have that WrappedMessageTypeMapper? Just pondering ...

Reply:
I think only the query context should have the `WrappedMessageTypeMapper` as that is what's required for client backwards compatibility. AFAIK there is no reason for a user to try to persist a QueryRequest, QueryResponse  etc. So I would say that the two contexts should remain independent.

Reply:
Certainly agreed for today. But do we anticipate any need for the persistence serctx to have knowledge of the types defined in ProtobufMetadataManager ?

Reply:
I can't think of a situation where the persistence context would need just a schema and not a marshaller. If the user needs to be able to marshall types, then they should just configure the schema and marshaller via the SerializationConfiguration.

Reply:
Although, one problem that I have had to solve in a rather crude way on my !jbmar-server branch is how we can make the protostream transcoders take into account both contexts: https://github.com/ryanemerson/infinispan/blob/ISPN-10591-server/server/core/src/main/java/org/infinispan/server/core/dataconversion/ProtostreamTranscoder.java